pkg/plugin/restartable_process.go

```diff
@@ -80,8 +81,13 @@ func (p *restartableProcess) reset() error {
 //
 // Callers of resetLH *must* acquire the lock before calling it.
 func (p *restartableProcess) resetLH() error {
+	if p.resetFailures > 10 {
```

Comment:
are you thinking about including any sort of backoff for this?

Reply:
I haven't fully fleshed this out yet, but we can/should discuss. Ideally we can handle the following scenarios:
1. process dies for whatever reason (e.g. oom kill), and restarting succeeds the first try
1. process dies for whatever reason, attempt(s) to restart fail early and quickly -> we surface the issue somehow (ark server shuts itself down, or we create events, or ?)

I would like to try to avoid the situation where the process dies, and it takes minutes or more to finish all the attempts to restart it.